core/src/main/java/hudson/slaves/Cloud.java

```diff
@@ -310,8 +317,68 @@ public HttpResponse doDoDelete() throws IOException {
         return new HttpRedirect("..");
     }
 
+
+    @Override
+    public boolean isNameEditable() {
+        return true;
+    }
+
+    @RequirePOST
+    @Restricted(NoExternalUse.class)
+    @Override
+    public HttpResponse doConfirmRename(@QueryParameter String newName) throws IOException, ServletException, Descriptor.FormException {
+        newName = newName == null ? null : newName.trim();
+        FormValidation validationError = doCheckNewName(newName);
+        if (validationError.kind != FormValidation.Kind.OK) {
+            throw new Failure(validationError.getMessage());
+        }
+        this.name = newName;
```

Comment:
Shouldn't we be doing something to persist this change to disk, the same way it would be persisted immediately if we changed it via the configure page?